src/index.ts

```diff
@@ -67,65 +63,84 @@ export function generateWords(nr: number, words: string[] = []): string {
     nr = Math.abs(nr);
   }
 
+  let remainder = 0;
+  let quotient: number;
+  let integer: number;
+  let decimals: number;
+
```

Comment:
I understand that javascript does hoisting but if you have to initialize variables do it at the top  of the function, or make a very good comment in the code for why they are here.

Reply:
no need to define them before getting over with the return conditions.

Reply:
As var, let, const declarations are automatically hoisted up to the top of the scope, it makes sense to place them at top of their scope so that you can read the code closer to how the interpreter will execute it.